uchroma/fxlib/reaction.py

```diff
@@ -0,0 +1,140 @@
+#
+# reaction.py - Copyright (C) 2017 Ryan Burns
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, version 3.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+# License for more details.
+#
+
+# pylint: disable=invalid-name
+
+from traitlets import Int, observe
+
+from uchroma.color import ColorUtils
+from uchroma.renderer import Renderer, RendererMeta
+from uchroma.traits import ColorTrait
+
+
+DEFAULT_SPEED = 6
+MAX_SPEED = 9
+EXPIRE_TIME_FACTOR = 0.25
+
+REACT_COLOR_KEY = 'reaction_color'
+
+
+class Reaction(Renderer):
+    """
+    Reaction creates a two tone animation effect based on the 'react' fx
+    Config Options:
+     - background_color: the static color when no keys are active
+     - color: the color the key will change to when pressed
+     - speed: (1 - 9) how fast the keys will change back. 9 is the fastest.
+    """
+    meta = RendererMeta('Reaction',
+                        'Keys change color when pressed',
+                        'Ryan Burns',
+                        '1.0')
+
+    # Configuration options
+    speed = Int(default_value=DEFAULT_SPEED, min=1, max=MAX_SPEED).tag(config=True)
+    background_color = ColorTrait().tag(config=True)
+    color = ColorTrait().tag(config=True)
+
+    def __init__(self, *args, **kwargs):
+        super(Reaction, self).__init__(*args, **kwargs)
+        self.fps = 30
+
+        # It seems like observers can be called before __init__
+        # How is this possible?
+        if not hasattr(self, 'init_speed'):
```

Comment:
Traitlets has some funny quirks, all the trait initialization (setting of default_value) happens via metaclass.